### PR TITLE
(#12660) Fix WC REST OAuth signature calculation

### PR DESCRIPTION
**Pull request by neonics**
Support array parameters with more than 1 level of nesting.
Sorting is deferred until the full parameter name is known.

### CODE REVIEW

**Comment by mikejolley** [comment]
@claudiosanches can you take a look when you're back?

**Comment by claudiosanches** [comment]
@mikejolley we should do as WP-API: https://github.com/WP-API/OAuth1/blob/master/lib/class-wp-rest-oauth1.php

**Comment by neonics** [comment]
FYI https://github.com/WP-API/OAuth1/blob/master/lib/class-wp-rest-oauth1.php#L675 also has the param sort bug.

**Comment by mikejolley** [comment]
Ok, in that case @claudiosanches, @neonics should report and patch this in the WP API, then we can update our version to match. 

Thats assuming they accept this as a bug. From the comments in https://github.com/woocommerce/woocommerce/issues/12660, it seems filter is not going to be supported in the API. We just build on top of that.

File: includes/api/class-wc-rest-authentication.php [hunk 1]
```diff
@@ -336,23 +336,24 @@ private function check_oauth_signature( $user, $params ) {
...
+				foreach ( self::flatten_value( $param_value ) as $param_key_inner => $param_value_inner ) {
+					$query_parameters[] = $param_key . $param_key_inner . '%3D' . $param_value_inner;
 				}
 			} else {
 				$query_parameters[] = $param_key . '%3D' . $param_value; // Join with equals sign.
 			}
 		}
+
+		// Sort parameters.
+		if ( ! usort( $query_parameters, 'strcmp' ) ) {
```
Comment (neonics) [inline includes/api/class-wc-rest-authentication.php]: typo: should be uksort

**Pull request closed without merging**